core/cache/watcher.go

```diff
@@ -25,3 +29,105 @@ type NotifyWatcher interface {
 	Watcher
 	Changes() <-chan struct{}
 }
+
+type notifyWatcherBase struct {
```

Comment:
Glad to see this, it became obvious it was needed as I was looking at watchers in the cache yesterday.  +1